### PR TITLE
Chore: Typecheck TypeScript in CI pipelines

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Type check our TypeScript code in CI pipelines.

### CODE REVIEW

File: scripts/circle-test-frontend.sh [hunk 1]
```diff
@@ -5,8 +5,10 @@ source "$(dirname "$0")/helpers/exit-if-fail.sh"
 
 start=$(date +%s)
 
-exit_if_fail npm run prettier:check
-exit_if_fail npm run test
+exit_if_fail yarn run prettier:check
+exit_if_fail yarn run typecheck
+exit_if_fail yarn run packages:typecheck
```
Comment (hugohaggmark) [inline scripts/circle-test-frontend.sh]: Maybe we should change the order of these two?

Comment (aknuds1) [reply scripts/circle-test-frontend.sh]: @hugohaggmark OK I'll do it - I have no idea which order is best tbqh :)

**Comment by aknuds1** [comment]
@hugohaggmark Can the front-end squad please fix the type checking errors in this branch?

**Comment by hugohaggmark** [comment]
@grafana/grafana-frontend-platform I'll look into these type errors.

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Pull request merged**